### PR TITLE
feat: implement configuration gui

### PR DESCRIPTION
**Pull request by maciejmajek**
## Purpose

RAI's configuration is mostly dictated by the config.toml file.

## Proposed Changes

This PR introduces a gui for configuring the following parts of RAI:
- Models
- Tracing
- ASR (not yet honored by RAI_ASR)
- TTS (not yet honored by RAI_TTS)
- Additional features

## Issues

- Links to relevant issues

## Testing

```sh
streamlit run src/rai/rai/utils/configurator.py
```

## Summary by CodeRabbit

- **New Features**
	- Introduced a Streamlit-based configuration wizard for setting up an AI environment, guiding users through model selection, tracing, speech recognition, and more.
	- Added new sections for automatic speech recognition (ASR) and text-to-speech (TTS) in the configuration, enhancing functionality.
	- Added a new section in the README for configuring RAI with updated setup instructions.
	- Included a new dependency for enhanced project management.

- **Bug Fixes**
	- Improved error handling in the configuration wizard to ensure necessary environment variables are present before saving configurations.

- **Refactor**
	- Updated the function for tracing callbacks to allow optional parameters for more flexible configuration.

- **Documentation**
	- Enhanced the README to clarify setup processes and limitations of the RAI framework, including resource usage concerns.

### CODE REVIEW

**Comment by maciejmajek** [comment]
@coderabbit pause

File: README.md [hunk 1]
```diff
@@ -106,6 +106,14 @@ rosdep install --from-paths src --ignore-src -r -y
 > poetry install --with openset,nomad
 > ```
 
+#### 1.4 Configure RAI
+
+Run the configuration tool to set up your vendor and other settings:
+
+```bash
+streamlit run src/rai/rai/utils/configurator.py
```
Comment (boczekbartek) [inline README.md]: `streamlit` is not visible without virtual env. But project is not built yes, so cannot use `setup_shell.sh`. So I think `poetry shell`.
```suggestion
```bash
poetry shell
streamlit run src/rai/rai/utils/configurator.py
```

File: README.md [hunk 2]
```diff
@@ -106,6 +106,14 @@ rosdep install --from-paths src --ignore-src -r -y
...
+```
+
```
Comment (boczekbartek) [inline README.md:117]: Add a sentence that user should go to the browser and open localhost:8501

File: src/rai/rai/utils/configurator.py [hunk 3]
```diff
@@ -0,0 +1,761 @@
...
+    st.session_state.config["asr"]["recording_device_name"] = recording_device_name
+    asr_vendor = st.selectbox(
+        "Choose your ASR vendor",
+        ["Local Whisper (Free)", "OpenAI (Cloud)"],
+        placeholder="Select vendor",
+    )
+    if asr_vendor == "Local Whisper (Free)":
+        st.info(
+            """
+        Recommended to use when Nvidia GPU is available.
```
Comment (boczekbartek) [inline src/rai/rai/utils/configurator.py]:
```suggestion
        Local Whisper is recommended to use when Nvidia GPU is available.
```

**Review by boczekbartek — CHANGES_REQUESTED** [review]
@maciejmajek Great feature! I left some comments. Also:
1. Would it be easy to make a drop-down list of available models? Or add a link to langchain/openai api website next to the text field?
2. For langsmith/cloud langfuse I would add a link to create accounts in the configurator
3. For langfuse I would use self-hosting (http://localhost:3000) as default and point the  user to this doc: https://langfuse.com/docs/deployment/local
4. I think saving doesn't work as expected:
```
[aws]
simple_model = "anthropic.claude-3-haiku-20240307-v1:0"
complex_model = "anthropic.claude-3-5-sonnet-20240620-v1:0"
embeddings_model = "amazon.titan-embed-text-v1"

[openai]
simple_model = "anthropic.claude-3-haiku-20240307-v1:0"
complex_model = "anthropic.claude-3-5-sonnet-20240620-v1:0"
embeddings_model = "amazon.titan-embed-text-v1"

[ollama]
simple_model = "anthropic.claude-3-haiku-20240307-v1:0"
complex_model = "anthropic.claude-3-5-sonnet-20240620-v1:0"
embeddings_model = "amazon.titan-embed-text-v1"
```

File: README.md (hunk 2 above)
Comment (maciejmajek) [reply README.md:117]: It opens automatically when run locally. The port is unknown before running the script. I will add a note that the links will be displayed
```bash
(rai-py3.12) mmajek:~/projects/rai$ streamlit run src/rai/rai/utils/configurator.py 

  You can now view your Streamlit app in your browser.

  Local URL: http://localhost:8503
  Network URL: http://xxx:8503
  External URL: http://xxx:8503
  ```

**Review by boczekbartek — APPROVED** [review]
Looks good!

**Pull request merged**